docs/DeveloperGuide.md

```diff
@@ -5,9 +5,261 @@
+- Check the 'isAvailable' attribute of the 'Book' instance to determine if it can be borrowed. 
+- If available, the book's 'borrowDate' is set to the current date, and 'returnDate' is calculated based
+on the pre-determined 2-week borrow period.
+- If not available, the user is informed of the 'returnDate'.
+- Post-interaction, 'BookStorage' updates the book's status in the storage file. 
+
+
+#### Return Command
+
+
+#### List Command
+##### Overview
+The "list command" is a feature that manages the listing all the books in the inventory.
+It is so librarians can keep track of all the books, as well as their status of whether they have
+been borrowed or are currently available in the library.
+
+##### Component-Level
+The "list command" component interfaces with several others:
+1. UI component : To relay messages back to the user.
+3. Book Domain Model : Represents the state and behaviour of the individual book entities.
+
+##### Class-Level
+1. Book Class : This class represents the domain entity with properties such as 'name', 'isAvailable',
+   'borrowDate' and 'returnDate' along with the methods to manipulate these properties.
+2. ListCommand Class : It handles the different commands of the list input, such as regular list,
+    list by alphabetical order, and so on.
+
+##### Implementation Details
+How? The "ListCommand" upon execution will:
+- Split user input with the regex "/sortby" to determine the various arguments the user has for the list function
+- If the user input is simply "list", the toString() function of each book in the ArrayList is called, printing out the books
+- If with argument, parses the argument to figure which it is, then creates a new ArrayList<Book> to copy
+    the original ArrayList and sort the new ArrayList according to the necessary argument. The toString()
+    function of each book in the newly sorted ArrayList is then called.
+- If there are no books in the original ArrayList, an exception is thrown and the user is informed of it.
+
+![ListCommandDiagram.png](images%2FListCommandDiagram.png)
+
+#### ListUser Command
+##### Overview
+The "listuser command" is a feature that allows the entire list of users who are people who have borrowed books
+along with all the books they borrowed 
+
+##### Component-Level
+The "listuser command" component interfaces with several others:
+1. UI component : To relay messages back to the user. 
+2. User Domain Model : Represents the state of the individual user entities.
+
+##### Class-Level
+1. User Class : This class represents the domain entity with properties such as 'name', 'borrowedBooks',
+   'unborrowBook' along with the methods to manipulate these properties.
+
+##### Implementation Details
+How? The "ListUserCommand" upon execution will:
+- Iterate through the list of users, by order of index
+- If list of users is empty, throws EmptyListException
+- For each user, iterates through the list of books and prints
+
```

Comment:
![Screenshot 2024-04-04 112508](https://github.com/nus-cs2113-AY2324S2/tp/assets/110823843/f1b15b48-de03-4904-814f-bbfcbe84393d)

the ListCommand diagram can't be seen, perhaps you could try reuploading it again?